### DISCUSSION TITLE
Are TransitionGuards needed?

### DISCUSSION THREAD

**Discussion by mtf90** [LearnLib/ralib#72] [AutomataLib Q & A] [answered]
Do we actually need the `TransitionGuard` as a standalone entity, or could the model be simplified if (guarded) `Transition`s were to directly hold a reference to the `GuardExpression`?

**Comment by pfg666** [comment, marked as answer]
Personally, I slightly prefer TransitionGuard being kept separate from GuardExpression, for the following reasons:

- it naturally captures a guard (defined by TransitionGuard) containing an expression (GuardExpression)
- we have, at least in the past, compared transition guards based on their identities, such comparison becomes problematic if we were to replace TransitionGuard by GuardExpressions (for which identity-comparison feels unintuitive)

**Reply by mtf90** [reply]
Fair enough, I can see the appeal especially of the second point.

**Comment by fhowar** [comment]
A second point would be that we may want to replace GuardExpressions by another implementation of expressions (e.g. constraints).
